broccoli/transforms/inject-node-globals.js

```diff
@@ -0,0 +1,52 @@
+function injectNodeGlobals({ types: t }) {
+  let requireId;
+  let importDecl;
+  let moduleId;
+  return {
+    name: 'inject require',
+    visitor: {
+      Program: {
+        enter(path, state) {
+          requireId = path.scope.globals.require;
+          moduleId = path.scope.globals.module;
+
+          if (requireId || moduleId) {
+
+            let specifiers = [];
+            let source = t.stringLiteral(this.file.resolveModuleSource('node-module'));
+
+            if (requireId) {
+              delete path.scope.globals.require;
+              specifiers.push(t.importSpecifier(requireId, t.identifier('require')));
+            }
+
+            if (moduleId) {
+              delete path.scope.globals.module;
+              specifiers.push(t.importSpecifier(moduleId, t.identifier('module')));
+            }
+
+            importDecl = t.importDeclaration(specifiers, source);
+            path.unshiftContainer('body', importDecl);
+          }
+
+        },
+        exit(path, state) {
+          if (requireId) {
+            path.scope.rename("require");
+          }
+        }
+      },
+      ImportDeclaration(path) {
+        if (path.node === importDecl) {
+          path.scope.registerDeclaration(path);
+        }
+      }
+    }
+  }
+}
+
+injectNodeGlobals.baseDir = function() {
+  return 'babel-plugin-transform-es2015-modules-amd';
```

Comment:
So it's going to do hash-for-dep. So `__dirname` is no good, this thing doesn't have any dependencies and deriving the cachkey from Ember's deps seems wrong.